### PR TITLE
Add workspace lifecycle page

### PR DESCRIPTION
**Pull request by loujaybee**
Add definitions of workspace statuses to the documentation. 

[**`Preview`**](https://deploy-preview-3279--gitpod-kumquat.netlify.app/docs/configure/workspaces/workspace-lifecycle)

Based upon: https://github.com/gitpod-io/website/pull/2858. 

Related to, but does not close: https://github.com/gitpod-io/website/issues/2368

image

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/3279"></a>

### CODE REVIEW

File: gitpod/docs/configure/workspaces/workspace-lifecycle.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+When you open a workspace, it will be in the "starting" state. This means that the workspace is being created and the initialization process is running.
+
+- Where a workspace is being provisioned and initialized.
+- If configured and available, a prebuild snapshot is used.
+- Otherwise, source control is downloaded into the workspace.
+
+### Workspace Running
+
+- An active container is provisioned within Gitpod.
```
Comment (loujaybee) [inline gitpod/docs/configure/workspaces/workspace-lifecycle.md:39]:
```suggestion
- An active workspace is provisioned within Gitpod.
```

File: gitpod/docs/configure/workspaces/workspace-lifecycle.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+- The workspace can be accessed by the user.
+
+### Workspace Stopped
+
+- No provisioned container is running (e.g. ports and URLs are not accessible).
```
Comment (loujaybee) [inline gitpod/docs/configure/workspaces/workspace-lifecycle.md:44]:
```suggestion
- No provisioned workspace is running (e.g. ports and URLs are not accessible).
```

File: gitpod/docs/configure/workspaces/workspace-lifecycle.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+### Workspace Snapshot
+
+You can create a snapshot of a workspace to save its state. This is useful if you want to keep a workspace around for a longer period of time, than the default. Read more about [Snapshots](/docs/configure/workspaces/collaboration).
+
+### Workspace Deleted
+
+Workspaces are deleted after 14 days. Pinned workspaces are never deleted automatically.
+
+<!-- TODO: Say how/where a user can see the cycle. -->
+<!-- TODO: Show screenshot of "workspace list" with helper info ("Workspaces that have been stopped for more than 24 hours. Inactive workspaces are automatically deleted after 14 days. Learn more") -->
```
Comment (loujaybee) [inline gitpod/docs/configure/workspaces/workspace-lifecycle.md:64]:
```suggestion
```

File: gitpod/docs/configure/workspaces/workspace-lifecycle.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+- The workspace can be accessed by the user.
+
+### Workspace Stopped
+
+- No provisioned workspace is running (e.g. ports and URLs are not accessible).
+- Only files and directories inside `/workspace` are preserved.
+- If the workspace is restarted, the URL is preserved.
+- A start is required before the workspace can be used.
+
+## Workspace Actions
```
Comment (loujaybee) [inline gitpod/docs/configure/workspaces/workspace-lifecycle.md:49]: Renamed as "Workspace Actions" (see [prior context](https://github.com/gitpod-io/website/pull/2858/files#r1042143278)), WDYT, André?

**Comment by loujaybee** [comment]
@axonasif, @atduarte, @gtsiolis -> As you were all quite heavily involved in the last PR (https://github.com/gitpod-io/website/pull/2858) if you're able to take a look through, leave any direct suggestions for changes, it'd be good to get this merged, and then we can incrementally bring other useful aspects from the previous PR.

File: gitpod/docs/configure/workspaces/workspace-lifecycle.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+## Workspace Statuses
+
+The state of the workspace is indicated by the color of the workspace indicator. For example, in the Gitpod [dashboard](https://gitpod.io/workspaces), workspace state is shown on the workspace list.
+
+1. 🟠 **Starting** - Workspace provisioning, inaccessible to the user.
+2. 🟢 **Running** - Workspace loaded, accessible to the user.
+3. 🟠 **Stopping** - Workspace being stopped, backups performing.
+4. 🔴 **Stopped** - Workspace no longer accessible. File system preserved for restart.
+
+> **Important:** Only files in the `workspace` directory are kept between state transitions.
```
Comment (atduarte) [inline gitpod/docs/configure/workspaces/workspace-lifecycle.md:23]: Important: 
```suggestion
> **Important:** Only files in the `/workspace` directory are kept between state transitions.
```

File: gitpod/docs/configure/workspaces/workspace-lifecycle.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+## Workspace Status Transitions
+
```
Comment (atduarte) [inline gitpod/docs/configure/workspaces/workspace-lifecycle.md:26]: Nit: As we are already inside the "Workspace Statuses" section and just described them more in-depth, I don't think there should be a separate "transitions" section. It seems we are calling "Statuses" and "Status Transitions" to the same thing.

```suggestion
```

File: gitpod/docs/configure/workspaces/workspace-lifecycle.md (hunk 4 above)
Comment (atduarte) [reply gitpod/docs/configure/workspaces/workspace-lifecycle.md:49]: Getting better, with a few caveats. The caveats don't do much harm, feel free to proceed if you would like:
1. I see Deleted as a Status. Similar to Terminated in AWS.
2. (Very nit) Given these are actions and not statuses, the continuous tense would sound better (e.g. Workspace Pinning).
3. I don't fully understand why these actions show up and others such as sharing don't 🤔

**Review by atduarte — APPROVED** [review]
Thank you, @loujaybee! Only one slightly important change request to avoid misunderstandings.

**Comment by Siddhant-K-code** [comment]
Looks good to :shipit: 

In follow-up PRs, we can take a look at [Timeout](https://github.com/gitpod-io/website/blob/2c9f6e74fea84ab3a75c2e790534e0aad7be70ba/gitpod/docs/configure/workspaces/workspace-lifecycle.md#timeouts) & other stuff also from our old PR

File: gitpod/docs/configure/workspaces/workspace-lifecycle.md (hunk 4 above)
Comment (loujaybee) [reply gitpod/docs/configure/workspaces/workspace-lifecycle.md:51]: 1. Makes sense, moved https://github.com/gitpod-io/website/pull/3279/commits/c81d4cb005c739f753dc29c6b07da83f7e5c13fb
2. Agree, updated! https://github.com/gitpod-io/website/pull/3279/commits/c81d4cb005c739f753dc29c6b07da83f7e5c13fb
3. Good point! This comes from the origin of the original issue / request, which was to support users understanding how workspaces load to support configuration, not necessarily to document everything workspace related, it just so happens that having these workspace statuses well documented supports the goal of helping with configuration. To that end, I'll probably proceed without the other "actions" but if you think a follow up makes sense, let's do it, and we can document any remaining workspace actions 🙏

**Comment by loujaybee** [comment]
> we can take a look at [Timeout](https://github.com/gitpod-io/website/blob/2c9f6e74fea84ab3a75c2e790534e0aad7be70ba/gitpod/docs/configure/workspaces/workspace-lifecycle.md#timeouts) & other stuff also from our old PR
Yep! I've got thoughts here and we need to cleanup this timeout section / page: https://www.gitpod.io/docs/introduction/learn-gitpod/the-life-of-a-workspace#timeouts

File: gitpod/docs/configure/workspaces/workspace-lifecycle.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+### Workspace Running
+
+- An active workspace is provisioned within Gitpod.
+- The workspace can be accessed by the user.
+
+### Workspace Stopped
+
+- No provisioned workspace is running (e.g. ports and URLs are not accessible).K
+- Only files and directories inside `/workspace` are preserved.
+- If the workspace is restarted, the URL is preserved.
```
Comment (akosyakov) [inline gitpod/docs/configure/workspaces/workspace-lifecycle.md:44]: there is no guarantee on it, URL is changed with each workspace deployment cc @loujaybee

File: gitpod/docs/configure/workspaces/workspace-lifecycle.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+- Otherwise, source control is downloaded into the workspace.
+
+### Workspace Running
+
+- An active workspace is provisioned within Gitpod.
+- The workspace can be accessed by the user.
+
+### Workspace Stopped
+
+- No provisioned workspace is running (e.g. ports and URLs are not accessible).K
```
Comment (akosyakov) [inline gitpod/docs/configure/workspaces/workspace-lifecycle.md:42]: typo `K`

File: gitpod/docs/configure/workspaces/workspace-lifecycle.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+> **Important:** Only files in the `/workspace` directory are kept between state transitions.
+
+The following describes each workspace status in detail, including what can cause a workspace to transition from one status to another.
+
+### Workspace Starting
+
+When you open a workspace, it will be in the "starting" state. This means that the workspace is being created and the initialization process is running.
+
+- Where a workspace is being provisioned and initialized.
```
Comment (akosyakov) [inline gitpod/docs/configure/workspaces/workspace-lifecycle.md:31]: image build is performed at this stage as well if the image was not built before

File: gitpod/docs/configure/workspaces/workspace-lifecycle.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+- If configured and available, a prebuild snapshot is used.
+- Otherwise, source control is downloaded into the workspace.
```
Comment (akosyakov) [inline gitpod/docs/configure/workspaces/workspace-lifecycle.md:33]: does it mean git downloaded ? 🤔 or that content is checked out using git?

File: gitpod/docs/configure/workspaces/workspace-lifecycle.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+
+## Workspace Actions
+
+### Workspace Pinning
+
+A pinned workspace is never deleted. You can pin a workspace from the Gitpod dashboard.
+
+### Workspace Snapshotting
+
+You can create a snapshot of a workspace to save its state. This is useful if you want to keep a workspace around for a longer period of time, than the default. Read more about [Snapshots](/docs/configure/workspaces/collaboration).
```
Comment (akosyakov) [inline gitpod/docs/configure/workspaces/workspace-lifecycle.md:59]: @loujaybee here is an intent to share reproducible setup with other, not to use it over a longer period of time. Usually restart is enough for that

File: gitpod/docs/configure/workspaces/workspace-lifecycle.md (hunk 7 above)
Comment (loujaybee) [reply gitpod/docs/configure/workspaces/workspace-lifecycle.md:44]: Ah good point about the generation part. I was referring to the ID, but you're right, it's ambiguous.

**Pull request merged**